### ISSUE TITLE
Feature Request (Windows 11): Improve "Hide Console" functionality to be compatible with Windows Terminal

### ISSUE THREAD

**Issue by EmulationFanatic** [Ryujinx/Ryujinx#3779]
#### Overview:
Windows 11 has a new Terminal app that is now on by default for Ryujinx (with Terminal settings set to "Let Windows decide" - the default setting), superseding the classic Windows console host. This breaks the "Hide Console" option in the GUI, with the console instead remaining open but minimized in the taskbar.

The fix might be:
1. Figure out a way to hide the Windows Terminal app and modify the function accordingly to behave the same regardless of which Terminal is called.
2. Force Ryujinx to call a Windows Console Host session specifically so that, when set to "Let Windows decide", it initiates a console host session. This doesn't seem like a viable option if someone has specifically selected the "Windows Terminal" option in Terminal Settings.

#### Smaller Details: 
The only current workaround is to go into Windows Settings > Privacy & Security > For developers > Terminal Settings and force Windows to use the Windows console host. This is unacceptable if someone wants to use the new Windows Terminal
![2022-10-20 07_49_50-Settings](https://user-images.githubusercontent.com/62343878/196968101-6610b1cf-5ca9-49b2-b0bf-d18a19dadb3a.png)

#### Nature of Request:  
[ Remove all that do not apply to your request. ]
- Change

### Why would this feature be useful?
To preserve the "Hide Console" menu option functionality in Windows 11.

**Comment by Squall-Leonhart** [comment]
Seems to be this
https://github.com/microsoft/terminal/issues/12570

**Comment by EmulationFanatic** [comment]
> Seems to be this [microsoft/terminal#12570](https://github.com/microsoft/terminal/issues/12570)

Yeah that is the behavior for sure. However it is still happening on versions of Windows Terminal newer than what they referenced when closing the issue, so there's more to the story yet.

**Comment by merryhime** [comment]
This is related to this change in May: https://github.com/microsoft/terminal/pull/13164, which only allows the ConPTY to minimize the console window only.

**Comment by r2rX** [comment]
> Unfortunately it doesn't appear to be a reasonable way to restore this functionality, and it would appear that Microsoft is very reluctantly supporting mixed TUI/GUI applications. Perhaps one would implement a custom console window in Avalonia instead?

This would be especially nice for Linux users, for those that launch ryujinx from their file explore/shortcut and not from a terminal.

**Comment by merryhime** [comment]
Just for future reference [System.Console.SetOut](https://learn.microsoft.com/en-us/dotnet/api/system.console.setout?view=net-6.0) looks like a viable redirection strategy.

**Comment by sidewinder94** [comment]
If Windows terminal is your default, whatever the reason. A workaround to get ryujinx to still use the "traditional" console host is to start ryujonx from the console host.

Easiest way to do that if you've installed ryujinx, is to change a shortcut used to launch it.

On the target field, add "conhost.exe" before the pull path to the ryyujinx executable.
You will have to change the icon afterwards, because it will then take the one of conhost.

I agree it's not a long term solution, but it should solve a couple issues.
